src/towncrier/test/test_build.py

```diff
@@ -836,3 +836,81 @@ def test_start_string(self):
         """)
 
         self.assertEqual(expected_output, output)
+
+    def test_uncommitted_files(self):
+        runner = CliRunner()
```

Comment:
It would be nice to have a dostring describing what feature this test tries to cover.

```suggestion

    def test_uncommitted_files(self):
        """
        At build time, it will delete any fragment file regardless of its stage,
        included files that are not part of the git reporsitory,
        or are just staged or modified.
        """
        runner = CliRunner()
```